src/main/java/org/icgc/argo/workflow_management/streams/WesConsumerConfig.java

```diff
@@ -71,70 +71,55 @@ public void init() {
+        .flatMap(this::consumeMessageAndExecuteInitializeOrCancel)
+        // Continue on error so disposable doesn't die.
+        // The tx commit/reject is handled in the flatmap so no need to worry about that here
+        .onErrorContinue(catchStreamError())
+        .log(WES_CONSUMER)
         .subscribe();
   }
 
-  private Consumer<Transaction<WfMgmtRunMsg>> consumeAndExecuteInitializeOrCancel() {
-    return tx -> {
-      val msg = tx.get();
-      log.debug("WfMgmtRunMsg received: {}", msg);
-
-      if (msg.getState().equals(RunState.INITIALIZING)) {
-        val runParams = createRunParams(msg);
-        wes.run(runParams)
-            .onErrorContinue(handleWesError(tx))
-            .subscribe(
-                runsResponse -> {
-                  log.info("Initialized: {}", msg);
-                  tx.commit();
-                });
-      } else if (msg.getState().equals(RunState.CANCELING)) {
-        wes.cancel(msg.getRunId())
-            .retryWhen(RetrySpec.backoff(3, Duration.ofMinutes(3)))
-            .onErrorContinue(handleWesError(tx))
-            .subscribe(
-                runsResponse -> {
-                  log.info("Cancelled: {}", runsResponse);
-                  tx.commit();
-                });
-      } else {
-        log.debug("Ignoring: {}", msg);
-        tx.commit();
-      }
-    };
+  private Mono<Void> consumeMessageAndExecuteInitializeOrCancel(Transaction<WfMgmtRunMsg> tx) {
+    val msg = tx.get();
+    log.debug("WfMgmtRunMsg received: {}", msg);
+
+    if (msg.getState().equals(RunState.INITIALIZING)) {
+      return wes.run(createRunParams(msg))
+          .flatMap(runsResponse -> commitTx("Initialized", tx))
+          .onErrorResume(t -> rejectAndWeblogTx(t, tx));
+    } else if (msg.getState().equals(RunState.CANCELING)) {
+      return wes.cancel(msg.getRunId())
+          .retryWhen(RetrySpec.backoff(3, Duration.ofMinutes(3)))
+          .flatMap(runsResponse -> commitTx("Cancelled", tx))
+          .onErrorResume(t -> rejectAndWeblogTx(t, tx));
+    } else {
+      return commitTx("Ignored", tx);
+    }
   }
 
-  private BiConsumer<Throwable, Object> handleRabbitStreamError() {
-    return (t, tx) -> {
-      t.printStackTrace();
-      log.error("Error occurred with: {}", tx);
-      if (tx instanceof Transaction<?> && ((Transaction<?>) tx).get() instanceof WfMgmtRunMsg) {
-        val msg = (WfMgmtRunMsg) ((Transaction<?>) tx).get();
-        msg.setState(RunState.SYSTEM_ERROR);
-        log.info("Rabbit stream SYSTEM_ERROR msg: {}", msg);
-        webLogEventSender.sendWfMgmtEventAsync(createWfMgmtEvent(msg));
-        ((Transaction<?>) tx).reject();
-      } else {
-        log.error("Transaction is lost, nothing to ack!");
-      }
-    };
+  private Mono<Void> commitTx(String actionMsg, Transaction<WfMgmtRunMsg> tx) {
+    log.info(actionMsg, tx.get());
+    tx.commit();
+    return Mono.empty();
```

Comment:
maybe return the transaction that was committed? ... does `commit()` return something?